src/thing_physics.c

```diff
@@ -662,6 +843,28 @@ TbBool thing_is_exempt_from_z_axis_clipping(const struct Thing *thing)
     }
     return false;
 }
+
+unsigned short get_slide_z_coord(const struct Thing *thing, const struct Coord3d *pos)
+// function at 451700. Original name unknown.
+{
+  unsigned short clipbox_size = thing->clipbox_size_yz;
+  long height = get_ceiling_height_above_thing_at(thing, pos);
+  short z_thing = (short)thing->mappos.z.val;
```

Comment:
Is it possible to move to `unsigned short` and remove cast?

Reply:
I think I tried that but it caused bugs.

Reply:
What bugs actually?

Reply:
Creatures kept going to the ceiling and dropping from it; basically, they "looped" through the floor and back through the ceiling. It seems there was some sort of overflow/underflow.

Reply:
Ok. This is sad but. ok